katdal/chunkstore.py

```diff
@@ -346,9 +354,9 @@ def put_dask_array(self, array_name, array):
         # Give better names to these two very similar variables
         in_name = array.name
         out_name = array_name
-        # Disambiguate out_name to avoid clashes in the dask graph
-        if out_name == in_name:
-            out_name = 'store-' + out_name
+        # Make out_name reasonably unique to avoid clashes and caches
+        token = da.core.tokenize(object())[:8]
```

Comment:
Looking at what `tokenize` does on `object`, it tries a bunch of things before falling back `uuid.uuid4().hex`; which then gets run through MD5. You might as well just use `uuid.uuid4().hex` and save the trip through MD5 (and the construction of a temporary object).

Also don't truncate it.

Reply:
I was trying to avoid the extra line of `import uuid` :-)

The truncation is on purpose to make the key more readable while still random enough. On the other hand, inspecting the graphs by eye is hopefully not a regular action, so............ OK.

Reply:
Done.